search/api.py

```diff
@@ -1,306 +1,142 @@
-        'agreed_to_terms_of_service': BOOL_TYPE,
-        'birth_city': NOT_ANALYZED_STRING_TYPE,
-        'birth_country': NOT_ANALYZED_STRING_TYPE,
-        'birth_state_or_territory': NOT_ANALYZED_STRING_TYPE,
-        'city': NOT_ANALYZED_STRING_TYPE,
-        'country': NOT_ANALYZED_STRING_TYPE,
-        'date_of_birth': DATE_TYPE,
-        'education': {'type': 'nested', 'properties': {
-            'degree_name': NOT_ANALYZED_STRING_TYPE,
-            'field_of_study': NOT_ANALYZED_STRING_TYPE,
-            'graduation_date': DATE_TYPE,
-            'id': LONG_TYPE,
-            'online_degree': BOOL_TYPE,
-            'school_city': NOT_ANALYZED_STRING_TYPE,
-            'school_country': NOT_ANALYZED_STRING_TYPE,
-            'school_name': NOT_ANALYZED_STRING_TYPE,
-            'school_state_or_territory': NOT_ANALYZED_STRING_TYPE,
-        }},
-        'email_optin': BOOL_TYPE,
-        'filled_out': BOOL_TYPE,
-        'first_name': NOT_ANALYZED_STRING_TYPE,
-        'gender': NOT_ANALYZED_STRING_TYPE,
-        'has_profile_image': BOOL_TYPE,
-        'last_name': NOT_ANALYZED_STRING_TYPE,
-        'preferred_language': NOT_ANALYZED_STRING_TYPE,
-        'preferred_name': NOT_ANALYZED_STRING_TYPE,
-        'pretty_printed_student_id': NOT_ANALYZED_STRING_TYPE,
-        'profile_url_full': NOT_ANALYZED_STRING_TYPE,
-        'profile_url_large': NOT_ANALYZED_STRING_TYPE,
-        'profile_url_medium': NOT_ANALYZED_STRING_TYPE,
-        'profile_url_small': NOT_ANALYZED_STRING_TYPE,
-        'username': NOT_ANALYZED_STRING_TYPE,
-        'work_history': {'type': 'nested', 'properties': {
-            'city': NOT_ANALYZED_STRING_TYPE,
-            'company_name': NOT_ANALYZED_STRING_TYPE,
-            'country': NOT_ANALYZED_STRING_TYPE,
-            'id': LONG_TYPE,
-            'industry': NOT_ANALYZED_STRING_TYPE,
-            'position': NOT_ANALYZED_STRING_TYPE,
-            'start_date': DATE_TYPE,
-            'state_or_territory': NOT_ANALYZED_STRING_TYPE,
-        }},
-    })
-    mapping.field("program", "object", properties={
-        'id': LONG_TYPE,
-        'grade_average': LONG_TYPE,
-        'enrollments': {'type': 'nested', 'properties': {
-            'course_details': {
-                'type': 'object',
-                'properties': {
-                    'course_modes': {
-                        'type': 'nested',
-                    }
-                }
-            }
-        }},
-        'certificates': {'type': 'nested'}
-    })
-    # Make strings not_analyzed by default
-    mapping.meta('dynamic_templates', [{
-        "notanalyzed": {
-            "match": "*",
-            "match_mapping_type": "string",
-            "mapping": NOT_ANALYZED_STRING_TYPE
-        }
-    }])
-    return mapping
+    def add_to_results(self, es_search_hits):
+        """
+        Adds a set of search results ('hits') to the combined results
+        """
+        self.results = self.results + list(es_search_hits)
```

Comment:
The `x = x + y` pattern has performance problems here because `x + y` creates a new list before assigning back to `x`. You may want to do `self.results += list(es_search_hits)` instead.